docs: Add initial mTLS setup (#200)
Substra/substra-documentation: docs/source/operations/howto/mtls-setup.rst

******************
How to set up mTLS
******************

> In this page we will describe how to set up mTLS communication between the backends and the orchestrator.

Reviewer: ```suggestion
This page describes how to set up mTLS communication between the backends and the orchestrator.
```

Author: Applied

In this scenario, the orchestrator will act as the certificate authority checking the certificates.
These instructions have to be repeated for each backend.

This guide assumes that you already have followed the instructions to :ref:`set up TLS<operations/howto/orchestrator-deployment:Setup TLS>`.

Generate backend Certificate Signing Request and signing key
============================================================

> The first step is to generate the Certificate Signing Request and a Signing key for the :term:`organization`.

Author: ```suggestion
The first step is to generate the Certificate Signing Request and a signing key for the :term:`organization`.
```

.. code:: bash

   openssl req -newkey rsa:2048 -nodes -keyout ORGNAME.key -subj "/O=ORGNAME/CN=HOSTNAME" -out ORGNAME.csr

| Replace ``ORGNAME`` with your :term:`organization` name.
  It should be the same as the value you put in your ``values.yaml`` file for the key ``orchestrator.mspID``.
| Replace ``HOSTNAME`` with the hostname of your substra backend.

> Then, you will need to send the file named ``ORGNAME.csr`` to the organization managing the orchestrator for them to sign your certificate.

Reviewer: ```suggestion
Then, you need to send the file named ``ORGNAME.csr`` to the organization managing the orchestrator for them to sign your certificate.
```

Signing the Substra backend certificate
=======================================

> Now that you have the Certificate Signing Request from your backend in your orchestrator, you can sign it with the orchestrator certificate authority.
> You need to navigate to the directory where are located the files ``orchestrator-ca.crt`` and ``orchestrator-ca.key`` (created during :ref:`TLS setup<operations/howto/orchestrator-deployment:Setup TLS>`).

Reviewer: ```suggestion
Now that you have the Certificate Signing Request from your backend in your orchestrator, you can sign it with the orchestrator certificate authority.
You need to navigate to the directory where the files ``orchestrator-ca.crt`` and ``orchestrator-ca.key`` are located (created during :ref:`TLS setup<operations/howto/orchestrator-deployment:Setup TLS>`).
```

Author: Done!

> Then you can sing the certificate with the following command:

Reviewer: ```suggestion
You can sign the certificate with the following command:
```

Author: Done!

| Replace ``ORGNAME`` with the :term:`organization` name.

.. caution::
    We don’t recommend having your certificate valid for a year (``365`` days in the previous command), you should change this value based on your company policy.

Then you need to send back the file named ``ORGNAME.crt`` to the organization managing the Substra backend. You don't need to keep a copy of this certificate.

Update backend configuration
============================

Once you received the certificate (named ``ORGNAME.crt``), you can create a secret in the Kubernetes cluster containing this file and the file ``ORGNAME.key`` with the following command:

.. code-block:: bash

   kubectl create secret tls orchestrator-client-cert --cert=ORGNAME.crt --key=ORGNAME.key

To use this certificate, you need to update or create the backend ``backend-values.yaml`` config file and add the following lines:

.. code-block:: yaml

   orchestrator:
     tls:
        enabled: true
        cacert: orchestrator-cacert
        mtls:
            enabled: true
            clientCertificate: orchestrator-client-cert

Note that you need to have the orchestrator TLS enabled for this to work.

> Once your config file is updated, you can either redeploy the backend to apply the changes or continue following the backend deployment guide.
> The backend can be updated with the following command:

Reviewer: ```suggestion
Once your config file is updated, you can either redeploy the backend to apply the changes or continue the backend deployment guide.
The backend can be updated with the following command:
```

Author: Done!

******************

In this page we will describe how to set up mTLS communication between the backends and the orchestrator.
In this scenario, the orchestrator will act as the certificate authority checking the certificates.
These instructions have to be repeated for each backend.

This guide assumes that you already have followed the instructions to :ref:`set up TLS<operations/howto/orchestrator-deployment:Setup TLS>`.

Generate backend Certificate Signing Request and signing key
============================================================

> The first step is to generate the Certificate Signing Request and a Signing key for the :term:`organization`.

Author: Signing (in Signing key) is capitalized here but not in the title
